cluster_integration.tf

```diff
@@ -11,7 +11,7 @@ resource "azuread_service_principal" "truefoundry_platform_features_service_prin
 
 resource "azuread_service_principal_password" "truefoundry_platform_features_service_principal_password" {
   count                = var.platform_feature_enabled ? var.feature_cluster_integration_enabled ? 1 : 0 : 0
-  service_principal_id = azuread_service_principal.truefoundry_platform_features_service_principal[0].object_id
+  service_principal_id = azuread_service_principal.truefoundry_platform_features_service_principal[0].id
```

Comment:
I don't think this is correct https://registry.terraform.io/providers/hashicorp/azuread/latest/docs/resources/service_principal#object_id 
Just check this one more time